### ISSUE TITLE
Rule proposal: `prefer-error-cause`

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#1782]
### Description

[Error#cause](https://2ality.com/2021/06/error-cause.html) is the new way to chain errors. Until now, the same thing was approximately achieved via various properties, like [`.innerError`](https://github.com/sindresorhus/serialize-error/blob/7e8be5138f59a31439134238de49c639343510ae/test.js#L128-L130) or `.error`.

What if there was a rule that detected all of these patterns and suggested using the new `cause`?

### Fail

```js
try {
	document.querySelector(':')
} catch (error) {
	const wrapped = new Error('x')
	wrapped.error = error;
	throw wrapped;
}
```

```js
class ExtendedError extends Error {
	constructor(message: string, error: Error) {
		super(message);
		this.innerError = error;
	}
}
```

### Pass

```js
try {
	document.querySelector(':')
} catch (error) {
	throw new Error('x', {cause: error})
}
```

### Additional Info

```js
class ExtendedError extends Error {
	constructor(message, {cause}) {
		super(message, {cause});
	}
}
// or

class ExtendedError extends Error {}
```

**Comment by fregante** [comment]
Exact duplicate of https://github.com/sindresorhus/eslint-plugin-unicorn/issues/1342 🤦‍♂️
